[Gradient Compression] Define a customized state for PowerSGD comm hook

wayi1 opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* **#48348 [Gradient Compression] Define a customized state for PowerSGD comm hook**

To support the features like error feedback, warm start, PowerSGD comm hook needs to maintain a state besides process group. Currently this state only includes a process group and a matrix approximation rank config.

This diff is a pure refactoring. Plan to add more state fields later.

Original PR issue: Investigate Applying PowerSGD to Communication Hook for Gradient Compression #47202

Differential Revision: [D25137962](https://our.internmc.facebook.com/intern/diff/D25137962/)

Commit:
[Gradient Compression] Define a customized state for PowerSGD comm hook

To support the features like error feedback, warm start, PowerSGD comm hook needs to maintain a state besides process group. Currently this state only includes a process group and a matrix approximation rank config.

This diff is a pure refactoring. Plan to add more state fields later.

Original PR issue: Investigate Applying PowerSGD to Communication Hook for Gradient Compression #47202

Differential Revision: [D25137962](https://our.internmc.facebook.com/intern/diff/D25137962/)

[ghstack-poisoned]

Commit:
Update on "[Gradient Compression] Define a customized state for PowerSGD comm hook"


To support the features like error feedback, warm start, PowerSGD comm hook needs to maintain a state besides process group. Currently this state only includes a process group and a matrix approximation rank config.

This diff is a pure refactoring. Plan to add more state fields later.

Original PR issue: Investigate Applying PowerSGD to Communication Hook for Gradient Compression #47202

Differential Revision: [D25137962](https://our.internmc.facebook.com/intern/diff/D25137962/)

[ghstack-poisoned]

Commit:
Update on "[Gradient Compression] Define a customized state for PowerSGD comm hook"


To support the features like error feedback, warm start, PowerSGD comm hook needs to maintain a state besides process group. Currently this state only includes a process group and a matrix approximation rank config.

This diff is a pure refactoring. Plan to add more state fields later.

Original PR issue: Investigate Applying PowerSGD to Communication Hook for Gradient Compression #47202

Differential Revision: [D25137962](https://our.internmc.facebook.com/intern/diff/D25137962/)

[ghstack-poisoned]

rohan-varma reviewed (approved):
LGTM! We should probably add unittesting for the fields when we add them, such as error feedback, warm start

Commit:
Update on "[Gradient Compression] Define a customized state for PowerSGD comm hook"


To support the features like error feedback, warm start, PowerSGD comm hook needs to maintain a state besides process group. Currently this state only includes a process group and a matrix approximation rank config.

This diff is a pure refactoring. Plan to add more state fields later.

Original PR issue: Investigate Applying PowerSGD to Communication Hook for Gradient Compression #47202

Differential Revision: [D25137962](https://our.internmc.facebook.com/intern/diff/D25137962/)

[ghstack-poisoned]
